netcam_rtsp.c

```diff
@@ -456,6 +452,23 @@ static int netcam_rtsp_open_context(netcam_context_ptr netcam){
         return -1;
     }
 
+    /*
+     * Documentation does not indicate thread safety on these
+     * init functions so we lock them just in case.
+     */
+    if (netcam->rtsp->status == RTSP_RECONNECTING){
+        pthread_mutex_lock(&global_lock);
+            avformat_network_deinit();
```

Comment:
I was interpreting the message that something associated with the loss of the camera has corrupted the network components of ffmpeg.  i.e. It flushed away at least parts of our previous initializations.  Cycling the network inits seemed the best option.  I think the risk is really just a race condition.  In that situation, I would think it would occur 100% of the time and would be evident in any testing but I didn't get any.  Do you have other suggestions?